routing/dht/routing.go

```diff
@@ -28,6 +29,8 @@ var asyncQueryBuffer = 10
 // PutValue adds value corresponding to given Key.
 // This is the top level "Store" operation of the DHT
 func (dht *IpfsDHT) PutValue(ctx context.Context, key key.Key, value []byte) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*30)
```

Comment:
The ctx is localized, not necessarily hardcoded.
So that it can be enforced globally (e.g. from a cfg file?).

One use case https://github.com/ipfs/go-ipfs/issues/1630 needs a long timeout for `GetValue` for a fix.